src/urls.ts

```diff
@@ -57,3 +57,45 @@ export const getWorkDetailsFromUrl = ({
     collectionName: url.match(/collections\/(\w+)/)?.[1],
   };
 };
+
+
+export const getPromptDetailsFromUrl = ({
+  // defining the input structure, eg.
+  // url: "https://archiveofourown.org/collections/mo_dao_zu_shi_kink_meme_2020/prompts/2644428"
+  url, 
+}: {
+  url: string;
+}): {
+  //defining the form of the return value/output
+  //collection name is the url name, not the Display Title
+  promptId: string;
+  collectionName: string
+} => {
+  const promptUrlMatch = url.match(/prompts\/(\d+)/);
```

Comment:
You might want to merge these in a single regex test, something like:

```ts
const collectionUrlMatch = url.match(/collections\/(\w+)/prompts\/(\d+)/);
if (!collectionUrlMatch) {
  throw new Error("Invalid prompt URL");
}
 
return {
  promptId: collectionUrlMatch[2],
  collectionName: collectionUrlMatch[1]
};
 ```
 
Double-check that I'm extracting the groups correctly, cause I always forget the syntax.